### PR TITLE
Add confirmation modal to proofreader assignment

### PR DESCRIPTION
**Pull request by joemull**
Part of https://github.com/BirkbeckCTP/janeway/issues/2753 .

### CODE REVIEW

**Comment by joemull** [comment]
I had to make one modification to implement this in the view logic: I moved the `unproofed_galleys` check into the confirmable form, since it fit the functionality quite well. The only difference is that while the earlier version would not let the user submit without having clicked the view or download buttons on each galley, this just warns you--you can still submit and override it. I don't see that as a problem, but let me know if you do.

File: views.py [hunk 1]
```diff
@@ -1240,19 +1241,17 @@ def typesetting_proofreading_assignment(request, assignment_id):
...
                 assignment.round.article,
             )
 
-        if 'complete' in request.POST:
-            unproofed_galleys = assignment.unproofed_galleys(galleys)
-            if not unproofed_galleys:
+        if form.is_valid():
+            form.save()
+
+            if form.is_confirmed():
```
Comment (mauromsl) [inline views.py:1254]: There is a slight change in behaviour here:

The check for unproofed galleys used to stop an author from completing an assignment. With this change, the author is warned but can still override the system. Nothing necessarily wrong  - I know you are a fan of user control, but just wanted to double check that this new behaviour is desired.

**Review by mauromsl — APPROVED** [review]
Nice use of the new form. I've added a comment in-line which I'm sure is something intentional but worth double checking.

File: views.py (hunk 1 above)
Comment (mauromsl) [reply views.py:1254]: Ah, I just saw this previous comment: https://github.com/BirkbeckCTP/typesetting/pull/160#issuecomment-1189074969

**Comment by ajrbyers** [comment]
> I had to make one modification to implement this in the view logic: I moved the `unproofed_galleys` check into the confirmable form, since it fit the functionality quite well and I couldn't handle it on the view with the new `is_confirmed` logic. The only difference is that while the earlier version would not let the user submit without having clicked the view or download buttons on each galley, this just warns you--you can still submit and override it. I don't see that as a problem, but let me know if you do.

It will be interesting to take a look and see how often files are left unproofed when users can skip the process.

**Pull request merged**